js/apps/system/_admin/aardvark/APP/frontend/js/views/documentsView.js

```diff
@@ -746,20 +755,24 @@
       } else {
         tooltip += ' (optional attribute, leave empty for autogenerated key)';
       }
-
+      var documentNameValidations = [
+        {
+          rule: Joi.string().regex(/^[a-zA-Z0-9_\-:\.@()\+,=;$!*\%'%]+$/),
+          msg: "Only these characters are allowed: a-z, A-Z, 0-9 and  _ - : . @ ( ) + , = ; $ ! * ' %."
+        },
+        {
+          rule: Joi.string().allow('').optional(),
+          msg: ''
+        }
```

Comment:
Same comments apply as above.

If the same rules appear repeatedly in the same file, I suggest putting them only in one place in this file, e.g. into a single function that returns them. That will reduce code maintenance long-term.

Reply:
Validation for key length (254 bytes max) is missing here, but is present in the other location in this file.

Reply:
moved validations to common helper